templates/admin/ppom-fields.php

```diff
@@ -244,22 +242,22 @@ class="ppom-modal-box ppom-slider ppom_sort_id_<?php echo esc_attr( $f_index );
 
 								<header>
 									<h3>
-										<?php echo $field_type; ?>
-										<span class="ppom-dataname-reader">(<?php echo $the_field_id; ?>)</span>
+										<?php echo esc_html( $field_type ); ?>
+										<span class="ppom-dataname-reader">(<?php echo esc_html( $the_field_id ); ?>)</span>
 									</h3>
 								</header>
 								<div class="ppom-modal-body">
 									<?php
-									echo $form_meta->render_field_meta( $defualt_fields, $field_type, $f_index, $field_meta );
+										echo $form_meta->render_field_meta( $defualt_fields, $field_type, $f_index, $field_meta ); // phpcs:ignore
```

Comment:
the [render_field_meta](https://github.com/Codeinwp/woocommerce-product-addon/blob/bugfix/pro/195/classes/fields.class.php#L300) method output does not have escaping.

I think we can make a refactor here.

Let's move the `echo` to inside of the `render_field_meta` do not return any value and remove the ignore announcement. Please make sure you've refactor all of the places which calls `render_field_meta`

Reply:
If we move echo to the inside of the render_field_meta it might issue here https://github.com/Codeinwp/woocommerce-product-addon/blob/bcbb51af430404e3c7ac8f49a081508287fd94ef/classes/fields.class.php#L202

And I've add phpcs:ignore becuase of `wp_kses_post` function automatically remove `class` from input tag
Ref: https://tinyurl.com/2afyk8j9 - It breaks the structure

Please let me know if you have any idea about this issue

Reply:
how about if we have echo in `render_field_settings` too?